### PR TITLE
iOS: Minor cleanup and warnings fixes

### PR DESCRIPTION
**Pull request by Adlai-Holler**
I was just scanning through the project the morning and saw a couple warnings and chances for improvement.

Cheers, love you guys!

Test plan: Run RNTester and see that it works.

### CODE REVIEW

File: React/Base/RCTJavaScriptLoader.mm [hunk 1]
```diff
@@ -202,7 +202,7 @@ + (NSData *)attemptSynchronousLoadOfBundleAtURL:(NSURL *)scriptURL
     return nil;
   }
   if (sourceLength) {
-    *sourceLength = statInfo.st_size;
+    *sourceLength = (NSUInteger)statInfo.st_size;
```
Comment (Adlai-Holler) [inline React/Base/RCTJavaScriptLoader.mm]: The changes in this file are to address an implicit loss of precision warning. I figured the best way was to make the loss of info explicit and standardize on NSUInteger as quickly as possible.

File: React/Modules/RCTUIManager.m [hunk 2]
```diff
@@ -768,7 +768,7 @@ - (void)_removeChildren:(NSArray<UIView *> *)children
     // So, let's temporary restore the view back after removing.
     // To do so, we have to memorize original `superview` (which can differ from `container`) and an index of removed view.
     UIView *originalSuperview = removedChild.superview;
-    NSUInteger *originalIndex = [originalSuperview.subviews indexOfObject:removedChild];
+    NSUInteger originalIndex = [originalSuperview.subviews indexOfObjectIdenticalTo:removedChild];
```
Comment (shergin) [inline React/Modules/RCTUIManager.m]: OMG, that's my fault.

**Comment by Adlai-Holler** [comment]
Hi @shergin, thanks for reviewing. I don't have the time-energy to split this diff at this time, so I'll reopen it in the future if I find some. Cheers!

**Comment by shergin** [comment]
😢 Anyways, thank you!

**Pull request closed without merging**